jujugui/static/gui/src/app/app.js

```diff
@@ -2199,6 +2197,11 @@ YUI.add('juju-gui', function(Y) {
           this._renderLogin();
           next();
           break;
+        case 'logout':
+          this._handleLogout();
+          this.state.changeState({root: 'login'});
```

Comment:
If we're in gisf the code below navigates away from the page. Could you add a comment here to that effect as it'll be easily missed in the future.